docs/ci.md

```diff
@@ -0,0 +1,134 @@
+For lint and unit test, the schematic diagram is as follows:
+
+```
++--------+     +--------+     +----------------+
+|        |     |        |     |                |
+|  root  +---->+  src   +---->+ lint/unit test |
+|        |     |        |     |                |
++--------+     +--------+     +----------------+
+```
+
+
+All the steps mentioned below are taking place inside a temporary work
+namespace.  When you click on the job details from your pull request, you can
+see the name of the work namespace in the dashboard UI.  The name starts with
+`ci-op-`.  The images created goes into this temporary work namespace.  At the
+end of every image build, the [work namespace name has set as an environment
+variable][namespace] called `OPENSHIFT_BUILD_NAMESPACE`.  This environment
+variable is used to refer the image URLs inside the configuration files.
+
+## root
+
+As part of the CI pipeline, the first step is to create the `root` image.  In
+fact, `root` is a tag created for the pipeline image.  This image contains all
+the tools including but not limited to Go compiler, git, kubectl, oc, and
+Operator SDK.
+
+The `root` image tag is created using this Dockerfile:
+`openshift-ci/Dockerfile.tools`.
+
+## src
+
+This step clones the pull request branch with the latest changes.  The cloning
+is taking place inside a container created from the `root` image.  As a result
+of this step, an image named `src` is going to be created.
+
+If your pull request depends on any package installed through `yum install`
+inside the `root` image, those changes should be sent through a different PR and
+merged first.  As you can see from the above diagrams, the `src` image gets
+built after the `root` image, whereas the pull request branch gets merged in the
+`src` image.
+
+## bin
+
+This step runs the `build` Makefile target.  This step is taking place
+inside a container created from the `src` image created in the
+previous step.
+
+The `make build` produces an operator binary image available under `./out`
+directory.  As a result of this step, a container image named `bin` is going to
+be created.
+
+## tests
+
+### lint
+
+The lint runs the [GolangCI Lint][golangci], [YAML Lint][yaml-lint], and
+[Operator Courier][operator-courier].  GolangCI is a Go program, whereas the
+other two are Python based.  So, Python 3 is a prerequisite to run lint.
+
+The GolangCI Lint program runs multiple Go lint tools against the repository.
+GolangCI Lint runs lint concurrently and completes execution in a few seconds.
+As of now, there is no configuration provided to run GolangCI Lint.
+
+The YAML Lint tools validate all the YAML configuration files.  It
+excludes the `vendor` directory while running.  There is a
+configuration file at the top-level directory of the source code:
+`.yamllint`.
+
+### unit
+
+The `unit` target runs the unit tests.  Some of the tests make use of mock
+objects. The unit tests don't require a dedicated OpenShift cluster, unlike
+end-to-end tests.
```

Comment:
We can add `It runs test-unit` Makefile target. (for example - mentioned in e2e below)

Reply:
Done. 048a77418d81a5799005c7203a4e344bb8cb7d24